src/protocol/vm/state.rs

```diff
@@ -533,28 +536,115 @@ impl VMPoolState<PreCachedDB> {
 
         merged
     }
+
+    async fn get_amount_out(
+        &self,
+        sell_token: ERC20Token,
+        sell_amount: U256,
+        buy_token: ERC20Token,
+    ) -> Result<(U256, U256, VMPoolState<PreCachedDB>), ProtosimError> {
+        let sell_amount_limit = self
+            .clone()
+            .get_sell_amount_limit(vec![sell_token.clone(), buy_token.clone()])
+            .await?;
+
+        let overwrites = self
+            .clone()
```

Comment:
Repeating my private message to you...

I don’t know what to do about the block lasting overwrites - since I can’t get rid of the self.clone without changing the entire interface of get_amount_out to take &mut self , which would affect even the native protocols. Seems like a big no-no to me, since we have already released the native protocols I guess? I’ve started looking into removing block_lasting_overwrites, but I think I don’t have enough brain power left for today to look into the implications of this. I can start with that tomorrow.

Reply:
Looks like we no longer need to clone here now?